database/influxdb/influxdb_database.py

```diff
@@ -0,0 +1,172 @@
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: influxdb_database
+short_description: Manage InfluxDB databases
+description:
+    - Manage InfluxDB databases
+version_added: 2.1
+author: "Kamil Szczygiel (@kamsz)"
+requirements:
+    - "python >= 2.6"
+    - "influxdb >= 0.9"
+options:
+    hostname:
+        description:
+            - The hostname or IP address on which InfluxDB server is listening
+        required: True
+    username:
+        description:
+            - Username that will be used to authenticate against InfluxDB server
+        default: root
+    password:
+        description:
+            - Password that will be used to authenticate against InfluxDB server
+        default: root
+    port:
+        description:
+            - The port on which InfluxDB server is listening
+        default: 8086
+    database_name:
+        description:
+            - Name of the database that will be created/destroyed
+        required: True
+    state:
+        description:
+            - Determines if the database should be created or destroyed
+        choices: ['present', 'absent']
+        default: present
+'''
+
+EXAMPLES = '''
+# Example influxdb_database command from Ansible Playbooks
+- name: Create database
+    influxdb_database:
+      hostname: "{{influxdb_ip_address}}"
+      database_name: "{{influxdb_database_name}}"
+      state: present
+
+- name: Destroy database
+    influxdb_database:
+      hostname: "{{influxdb_ip_address}}"
+      database_name: "{{influxdb_database_name}}"
+      state: absent
+
+- name: Create database using custom credentials
+    influxdb_database:
+      hostname: "{{influxdb_ip_address}}"
+      username: "{{influxdb_username}}"
+      password: "{{influxdb_password}}"
+      database_name: "{{influxdb_database_name}}"
+      state: present
+'''
+
+RETURN = '''
+#
+'''
+
+import requests.exceptions
+try:
+    from influxdb import InfluxDBClient
+    from influxdb import exceptions
+    HAS_INFLUXDB = True
+except ImportError:
+    HAS_INFLUXDB = False
+
+
+def influxdb_argument_spec():
+    return dict(
+        hostname=dict(required=True),
+        port=dict(default=8086, type='int'),
+        username=dict(default='root'),
+        password=dict(default='root', no_log=True),
+        database_name=dict(required=True)
+    )
+
+
+def connect_to_influxdb(module):
+    hostname = module.params['hostname']
+    port = module.params['port']
+    username = module.params['username']
+    password = module.params['password']
+    database_name = module.params['database_name']
+
+    client = InfluxDBClient(
+        host=hostname,
+        port=port,
+        username=username,
+        password=password,
+        database=database_name
+    )
+    return client
+
+
+def create_database(module):
+    database_name = module.params['database_name']
+    client = connect_to_influxdb(module)
+
+    if not module.check_mode:
```

Comment:
@kamsz Noticed that you're not checking if the database already exists here. Like in the retention policy module, the check mode conditional should go inside the try block while the change detection goes to the outside.